static/app/views/starfish/queries/useFullSpanFromTrace.tsx

```diff
@@ -0,0 +1,36 @@
+import {useEventJSON} from 'sentry/views/starfish/queries/useEventJSON';
+import {useIndexedSpans} from 'sentry/views/starfish/queries/useIndexedSpans';
+import {SpanIndexedFields} from 'sentry/views/starfish/types';
+
+// NOTE: Fetching the top one is a bit naive, but works for now. A better
+// approach might be to fetch several at a time, and let the hook consumer
+// decide how to display them
+export function useFullSpanFromTrace(group?: string, enabled: boolean = true) {
+  const filters: {[key: string]: string} = {};
+
+  if (group) {
+    filters[SpanIndexedFields.SPAN_GROUP] = group;
+  }
+
+  const indexedSpansResponse = useIndexedSpans(filters, 1, enabled);
```

Comment:
should we set enabled to false if group is undefined so we don't run this query?

Reply:
Great Q, I decided not to for two reasons:

1. If there's an explicit "enabled" argument, it would be surprising if the component decided to override it with other logic based on other arguments
2. `group` is just a filter like any other (the only one I support in this hook, unfortch) so it'd be weird if a missing group prevented the hook from firing. If the hook supported other filters (by action, or domain, or whatever, which it will eventually) the `group` filter would no longer be special

Maybe I should just add a couple more filters right now, so the hook makes more sense?

Reply:
Hm, yeah makes sense, especially if you're going to be using this hook with other filters!